docs/tutorials/docker/getting-started.md

```diff
@@ -1,198 +1,297 @@
+After you have installed Docker on your machine, you should be able to pull the Docker image down from quay.io:
 
-      [[eosio::action]]
-      void world( eosio::name name ) {
-        print("Hi, ", name);
-      }
-};
+```sh
+docker pull quay.io/ultra.io/3rdparty-devtools:latest
 ```
 
-### Compiling a Smart Contract
+You will also want to use this command to update the Docker image you have locally.
+
+The Docker image we provide is located at [our official quay.io repository](https://quay.io/ultra.io/3rdparty-devtools). If you are interested, you can check there to see if there is a new version of the image available, or use an older version if needed.
+
+## Running the Developer Tools Docker image
 
-Inside of the docker image (using the terminal that is attached to the `ultra` container) navigate into the `contracts` directory, and run the following command.
+To start the Docker container with the cloned Developer Tools Docker image use _one_ of the following commands appropriate for your operating system:
 
+::: details Windows
+```sh
+docker run -dit --name ultra -p 8888:8888 -p 9876:9876 -v %cd%/ultra_workdir:/opt/ultra_workdir --name ultra quay.io/ultra.io/3rdparty-devtools:latest
 ```
-mkdir -p /opt/ultra_workdir/contracts && cd /opt/ultra_workdir/contracts
+
+You will be asked if you are okay with sharing the working directory with the Docker container. Only the content of the directory specified will be shared, and not any other content of your drive.
+
+![](./images/docker-windows-allow-sharing.png)
+:::
+
+::: details Linux
+```sh
+docker run -dit --name ultra -p 8888:8888 -p 9876:9876 -v ~/ultra_workdir:/opt/ultra_workdir --name ultra quay.io/ultra.io/3rdparty-devtools:latest
 ```
+:::
+
+After you created the container you realistically won't need to create it again. Existing container will be accessible under the name of `ultra`.
+
+::: info
+If you encounter an error: `Error response from daemon: Conflict. The container name "/ultra" is already in use`, then it means you already have the container running and don't need to do anything.
+:::
+
+::: warning
+The above command will utilize ports 8888 and 9876. If those ports are occupied the docker will fail to create the container. In case you encounter this issue, make sure you don't already have a container running with those ports in use. Otherwise, you can adjust the first number in port publishing (e.g. `8889:8888` instead of `8888:8888`) or remove the port publishing arguments completely (e.g. `-p 8888:8888 -p 9876:9876`).
+:::
 
+To stop the container without destroying it you can use the following command. Stopping a container closes all running applications.
+
+```sh
+docker stop ultra
 ```
-cdt-cpp -abigen -o hello.wasm hello.cpp
+
+## Accessing the image
+
+After you created the container you will be able to attach to it using the following command. It will also start the container if it is currently stopped.
+
+```sh
+docker start ultra && docker attach ultra
```

Comment:
This command may lead to unintended behavior. If you attach to the container using docker attach ultra and then log out, the container will stop. To keep your container running after you log out, use the following command instead:

```sh
docker exec -it ultra /bin/bash
```
This way, you can interact with the container without causing it to stop upon exit.